File: cpp/643_子数组最大平均数I_easy.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int n = nums.size();
        int sum = 0;
        for (int i=0; i<k; i++) sum += nums[i];
        int maxsum = sum;
        for (int i=k; i<n; i++){
            sum = sum + nums[i] - nums[i-k];
            maxsum = max(maxsum, sum);
        }
        return (long long)maxsum / (k*1.0);
    }
};

File: python/643_子数组最大平均数I_easy.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        tmp = sum(nums[:k])
        res = tmp
        for i in range(k, len(nums)):
            tmp = tmp + nums[i] - nums[i-k]
            res = max(res, tmp)
        return res/k